[dynamo] Support dynamic slicing

tugsbayasgalan opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #91341



cc @mlazos @soumith @voznesenskym @yanboliang @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @chunyuan-w @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @desertfire

tugsbayasgalan committed:
[dynamo] Support dynamic slicing

[ghstack-poisoned]

tugsbayasgalan commented:
@tugsbayasgalan has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D42223259).

tugsbayasgalan commented:
@pytorchbot rebase

tugsbayasgalan committed:
Update on "[dynamo] Support dynamic slicing"




cc mlazos soumith voznesenskym yanboliang penguinwu anijain2305 EikanWang jgong5 Guobing-Chen chunyuan-w XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx desertfire

Differential Revision: [D42223259](https://our.internmc.facebook.com/intern/diff/D42223259)

[ghstack-poisoned]

tugsbayasgalan committed:
Update on "[dynamo] Support dynamic slicing"




cc mlazos soumith voznesenskym yanboliang penguinwu anijain2305 EikanWang jgong5 Guobing-Chen chunyuan-w XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx desertfire

Differential Revision: [D42223259](https://our.internmc.facebook.com/intern/diff/D42223259)

[ghstack-poisoned]

tugsbayasgalan commented:
@tugsbayasgalan has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D42223259).

voznesenskym commented on torch/_dynamo/variables/lists.py:
```diff
@@ -451,15 +448,18 @@ def __init__(self, items, **kwargs):
+
+        # TODO (tugsbayasgalan) Address this later, the implementation
+        # of slice unpacking does some adjustments based on step size
+        # to start and stop. It is bit unclear what those adjustment
+        # be when step is symbolic
```
> what those adjustments be 

did you forget a word?

tugsbayasgalan replied:
oops this is addressed. Let me publish a new version.

voznesenskym commented on test/dynamo/test_repros.py:
```diff
@@ -2226,6 +2227,79 @@ def f(x):
+
+        self.assertEqual(count, 2)
+
+        gm_dynamo_mode, _ = torch._dynamo.export(f, torch.randn(4, 5), aten_graph=False)
```
nit: torch_mode if the other is aten_mode ;)

voznesenskym replied:
Would be cool to also assure the guards come out sane?

voznesenskym commented on test/dynamo/test_repros.py:
```diff
@@ -2226,6 +2227,79 @@ def f(x):
+        # one for x.shape[0]-2 and one for x.shape[1]-1 and one for slice
+        count = 0
+        for node in gm_dynamo_mode.graph.nodes:
+            if node.op == "call_function" and node.target == operator.getitem:
+                count += 1
```
The logic differs a bit, because the representation of lookup differs - but it makes this test confusing. Add a comment?

voznesenskym reviewed (approved):
LGTM, lots of nits, please address

tugsbayasgalan committed:
Update on "[dynamo] Support dynamic slicing"




cc mlazos soumith voznesenskym yanboliang penguinwu anijain2305 EikanWang jgong5 Guobing-Chen chunyuan-w XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx desertfire

Differential Revision: [D42223259](https://our.internmc.facebook.com/intern/diff/D42223259)

[ghstack-poisoned]

tugsbayasgalan commented:
@tugsbayasgalan has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D42223259).

tugsbayasgalan commented:
@tugsbayasgalan has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D42223259).

tugsbayasgalan commented:
@pytorchbot rebase

tugsbayasgalan committed:
Update on "[dynamo] Support dynamic slicing"




cc mlazos soumith voznesenskym yanboliang penguinwu anijain2305 EikanWang jgong5 Guobing-Chen chunyuan-w XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx desertfire

Differential Revision: [D42223259](https://our.internmc.facebook.com/intern/diff/D42223259)

[ghstack-poisoned]

tugsbayasgalan committed:
Update on "[dynamo] Support dynamic slicing"




cc mlazos soumith voznesenskym yanboliang penguinwu anijain2305 EikanWang jgong5 Guobing-Chen chunyuan-w XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx desertfire

Differential Revision: [D42223259](https://our.internmc.facebook.com/intern/diff/D42223259)

[ghstack-poisoned]

tugsbayasgalan committed:
Update on "[dynamo] Support dynamic slicing"




cc mlazos soumith voznesenskym yanboliang penguinwu anijain2305 EikanWang jgong5 Guobing-Chen chunyuan-w XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx desertfire

Differential Revision: [D42223259](https://our.internmc.facebook.com/intern/diff/D42223259)

[ghstack-poisoned]

tugsbayasgalan commented:
@tugsbayasgalan has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D42223259).

tugsbayasgalan committed:
Update on "[dynamo] Support dynamic slicing"




cc mlazos soumith voznesenskym yanboliang penguinwu anijain2305 EikanWang jgong5 Guobing-Chen chunyuan-w XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx desertfire

Differential Revision: [D42223259](https://our.internmc.facebook.com/intern/diff/D42223259)

[ghstack-poisoned]

tugsbayasgalan committed:
Update on "[dynamo] Support dynamic slicing"




cc mlazos soumith voznesenskym yanboliang penguinwu anijain2305 EikanWang jgong5 Guobing-Chen chunyuan-w XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx desertfire

Differential Revision: [D42223259](https://our.internmc.facebook.com/intern/diff/D42223259)

[ghstack-poisoned]

tugsbayasgalan committed:
Update on "[dynamo] Support dynamic slicing"




cc mlazos soumith voznesenskym yanboliang penguinwu anijain2305 EikanWang jgong5 Guobing-Chen chunyuan-w XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx desertfire

Differential Revision: [D42223259](https://our.internmc.facebook.com/intern/diff/D42223259)

[ghstack-poisoned]

tugsbayasgalan commented:
@tugsbayasgalan has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D42223259).

tugsbayasgalan committed:
Update on "[dynamo] Support dynamic slicing"




cc mlazos soumith voznesenskym yanboliang penguinwu anijain2305 EikanWang jgong5 Guobing-Chen chunyuan-w XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx desertfire

Differential Revision: [D42223259](https://our.internmc.facebook.com/intern/diff/D42223259)

[ghstack-poisoned]

tugsbayasgalan committed:
Update on "[dynamo] Support dynamic slicing"




cc mlazos soumith voznesenskym yanboliang penguinwu anijain2305 EikanWang jgong5 Guobing-Chen chunyuan-w XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx desertfire

Differential Revision: [D42223259](https://our.internmc.facebook.com/intern/diff/D42223259)

[ghstack-poisoned]

tugsbayasgalan commented:
@tugsbayasgalan has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D42223259).

tugsbayasgalan committed:
Update on "[dynamo] Support dynamic slicing"




cc mlazos soumith voznesenskym yanboliang penguinwu anijain2305 EikanWang jgong5 Guobing-Chen chunyuan-w XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx desertfire

Differential Revision: [D42223259](https://our.internmc.facebook.com/intern/diff/D42223259)

[ghstack-poisoned]

tugsbayasgalan committed:
Update on "[dynamo] Support dynamic slicing"




cc mlazos soumith voznesenskym yanboliang penguinwu anijain2305 EikanWang jgong5 Guobing-Chen chunyuan-w XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx desertfire

Differential Revision: [D42223259](https://our.internmc.facebook.com/intern/diff/D42223259)

[ghstack-poisoned]

tugsbayasgalan commented:
@tugsbayasgalan has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D42223259).

tugsbayasgalan committed:
Update on "[dynamo] Support dynamic slicing"




cc mlazos soumith voznesenskym yanboliang penguinwu anijain2305 EikanWang jgong5 Guobing-Chen chunyuan-w XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx desertfire

Differential Revision: [D42223259](https://our.internmc.facebook.com/intern/diff/D42223259)

[ghstack-poisoned]

clee2000 commented:
@pytorchbot revert -m "breaking various tests https://hud.pytorch.org/pytorch/pytorch/commit/8e7dcd140ace26a7e3096a26fbeec9f572e9aaa7 https://github.com/pytorch/pytorch/actions/runs/3856936505/jobs/6574089745 marking this as weird because it was merged via codev?" -c weird

tugsbayasgalan committed:
Update on "[dynamo] Support dynamic slicing"




cc mlazos soumith voznesenskym yanboliang penguinwu anijain2305 EikanWang jgong5 Guobing-Chen chunyuan-w XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx desertfire

Differential Revision: [D42223259](https://our.internmc.facebook.com/intern/diff/D42223259)

[ghstack-poisoned]

tugsbayasgalan commented:
@pytorchbot rebase

tugsbayasgalan commented:
@pytorchbot rebase

ezyang committed:
Update on "[dynamo] Support dynamic slicing"




cc mlazos soumith voznesenskym yanboliang penguinwu anijain2305 EikanWang jgong5 Guobing-Chen chunyuan-w XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx desertfire


[ghstack-poisoned]

ezyang commented:
@pytorchbot merge

tugsbayasgalan commented:
@pytorchbot rebase

ezyang committed:
Update on "[dynamo] Support dynamic slicing"




cc mlazos soumith voznesenskym yanboliang penguinwu anijain2305 EikanWang jgong5 Guobing-Chen chunyuan-w XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx desertfire


[ghstack-poisoned]

tugsbayasgalan commented:
@pytorchbot merge

tugsbayasgalan committed:
Update on "[dynamo] Support dynamic slicing"




cc mlazos soumith voznesenskym yanboliang penguinwu anijain2305 EikanWang jgong5 Guobing-Chen chunyuan-w XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx desertfire


[ghstack-poisoned]

ezyang commented:
@pytorchbot merge

tugsbayasgalan commented:
Looks like it incorrectly thinks internal change has been landed already??
